var/spack/repos/builtin/packages/picard/package.py

```diff
@@ -169,7 +171,12 @@ def install(self, spec, prefix):
 
         # Set up a helper script to call java on the jar file,
         # explicitly codes the path for java and the jar file.
-        script_sh = join_path(os.path.dirname(__file__), "picard.sh")
+
+        script_sh = join_path(
+            os.path.dirname(__file__),
+            "picard_with_parameters.sh" if "+parameters" in self.variants else "picard.sh",
+        )
```

Comment:
This doesn't work.  You want to use the proper check (see https://spack.readthedocs.io/en/latest/packaging_guide.html#boolean-variants).
```suggestion
        script_sh = join_path(
            os.path.dirname(__file__),
            "picard_with_parameters.sh" if "+parameters" in spec else "picard.sh",
        )
```